src/cocotb_tools/runner.py

```diff
@@ -1506,6 +1506,112 @@ def _test_command(self) -> List[_Command]:
         return cmds
 
 
+class Vcs(Runner):
+    """Implementation of :class:`Runner` for VCS.
+
+    * Does not support the ``pre_cmd`` argument to :meth:`.test`.
+    * Does not support VHDL.
+    """
+
+    supported_gpi_interfaces = {"verilog": ["vpi"]}
+
+    @staticmethod
+    def _simulator_in_path() -> None:
+        if shutil.which("vcs") is None:
+            raise SystemExit("ERROR: vcs executable not found!")
+
+    @staticmethod
+    def _get_include_options(includes: Sequence[PathLike]) -> _Command:
+        return [f"+incdir+{include}" for include in includes]
+
+    @staticmethod
+    def _get_define_options(defines: Mapping[str, object]) -> _Command:
+        return [f"+define+{name}={value}" for name, value in defines.items()]
+
+    def _get_parameter_options(self, parameters: Mapping[str, object]) -> _Command:
+        assert self.hdl_toplevel is not None
+        return [
+            f"-pvalue+{self.hdl_toplevel}.{name}={value}"
+            for name, value in parameters.items()
+        ]
+
+    @property
+    def pli_table_file(self) -> Path:
+        return self.build_dir / "pli.tab"
+
+    @property
+    def pli_table(self) -> Path:
```

Comment:
Be mindful of what's public/private. Does this or the methods/properties below need to be public?

Reply:
I have left `build_opts` with intention to be overloaded in custom version. Should it be `_build_opts`?

Reply:
Well seeing as all our other overloadable functions are private as well, I think so.